apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/amqp_events_producer.ex

```diff
@@ -45,70 +42,55 @@ defmodule Astarte.DataUpdaterPlant.AMQPEventsProducer do
   # Server callbacks
 
   def init(_args) do
-    rabbitmq_connect(false)
-  end
-
-  def terminate(_reason, %Channel{conn: conn} = chan) do
-    Channel.close(chan)
-    Connection.close(conn)
+    rabbitmq_connect()
   end
 
-  def handle_call({:publish, exchange, routing_key, payload, opts}, _from, chan) do
-    reply = Basic.publish(chan, exchange, routing_key, payload, opts)
-
-    {:reply, reply, chan}
+  def handle_call(
+        {:publish, exchange, routing_key, payload, opts},
+        _from,
+        state
+      ) do
+    ExRabbitPool.with_channel(:producers_pool, fn {:ok, channel} ->
+      reply = Basic.publish(channel, exchange, routing_key, payload, opts)
+      {:reply, reply, state}
+    end)
   end
 
-  def handle_call({:declare_exchange, exchange}, _from, chan) do
+  def handle_call({:declare_exchange, exchange}, _from, state) do
     # TODO: we need to decide who is responsible of deleting the exchange once it is
     # no longer needed
-    reply = Exchange.declare(chan, exchange, :direct, durable: true)
-
-    {:reply, reply, chan}
+    ExRabbitPool.with_channel(:producers_pool, fn {:ok, channel} ->
+      reply = Exchange.declare(channel, exchange, :direct, durable: true)
+      {:reply, reply, state}
+    end)
   end
 
   def handle_info(:try_to_connect, _state) do
```

Comment:
This is actually unused now so it can be removed

Reply:
Is it? I see `declare_exchange` being called [here](https://github.com/astarte-platform/astarte/blob/ffad806921d493a4cf4eb7d3105acf423c28ac9b/apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/triggers_handler.ex#L42)

Reply:
I was talking about `:try_to_connect`